### PR TITLE
Implement Zoombox

### PR DESCRIPTION
**Pull request by Equbuxu**
A replacement for the zoom box from extendedwpftoolkit.

### CODE REVIEW

**Review by CPKreuz — APPROVED** [review]
Works with stylus

**Review by flabbet — CHANGES_REQUESTED** [review]
Bug: Make canvas 16x16 then create new document with size 9999x9999, zoombox forbids to see whole canvas

File: PixiEditor/Views/UserControls/Zoombox.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+            {
+                prevMousePos = e.GetPosition(parent.mainCanvas);
+                parent.mainCanvas.CaptureMouse();
+            }
+
+            public void Update(MouseEventArgs e)
+            {
+                var curMousePos = e.GetPosition(parent.mainCanvas);
+                parent.SpaceOriginPos += curMousePos - prevMousePos;
+                prevMousePos = e.GetPosition(parent.mainCanvas);
```
Comment (CPKreuz) [inline PixiEditor/Views/UserControls/Zoombox.xaml.cs:46]: Wouldn't `curMousePos` already be assigned to the correct point

**Comment by flabbet** [comment]
Also, resizing existing canvas doesn't recenter zoombox

File: PixiEditor/Views/UserControls/Zoombox.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+            activeDragOperation.Start(e);
+        }
+
+        private void OnMouseUp(object sender, MouseButtonEventArgs e)
+        {
+            if (e.ChangedButton == MouseButton.Right)
+                return;
+            if (activeDragOperation != null)
+            {
+                activeDragOperation?.Terminate();
```
Comment (CPKreuz) [inline PixiEditor/Views/UserControls/Zoombox.xaml.cs:246]: No null check is required here as it's already done 2 lines above

**Review by CPKreuz — CHANGES_REQUESTED** [review]

File: PixiEditor/Views/UserControls/Zoombox.xaml.cs (hunk 1 above)
Comment (Equbuxu) [reply PixiEditor/Views/UserControls/Zoombox.xaml.cs:46]: I guess

File: PixiEditor/Views/UserControls/Zoombox.xaml.cs (hunk 2 above)
Comment (Equbuxu) [reply PixiEditor/Views/UserControls/Zoombox.xaml.cs:246]: yep

**Review by flabbet — CHANGES_REQUESTED** [review]
Bug: Create canvas 16x16, then Create new document with size 500x16, canvas is not recentered

**Review by flabbet — APPROVED** [review]
Everything seems to work

**Review by CPKreuz — APPROVED** [review]

**Pull request merged**